### PR TITLE
update connexion 3

### PR DESCRIPTION
**Pull request by petrblaho**
- chore: updates connexion to 3.x version
- chore: Connexion 3.0 update related changes
- test: adds missing content type to a post call
- fix: relaxes validation for response content

RHINENG-2950

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: tests/test_v1_api.py [hunk 1]
```diff
@@ -45,7 +45,11 @@ def test_comparison_report_api_post(self):
                 "11b3cbce-25a9-11e9-8457-c85b761454fa",
             ]
         }
-        response = self.client.post("api/drift/v1/comparison_report", data=data)
+        response = self.client.post(
+            "api/drift/v1/comparison_report",
+            data=data,
+            headers={"Content-Type": "application/json"},
```
Comment (vkrizan) [inline tests/test_v1_api.py:51]: Is there a constant somewhere for the type?

File: drift/app.py [hunk 2]
```diff
@@ -22,7 +22,8 @@ def create_app():
     app_name = os.getenv("APP_NAME", "drift")
     openapi_args = {"path_prefix": config.path_prefix, "app_name": app_name}
     connexion_app = FlaskApp(__name__, specification_dir="openapi/", arguments=openapi_args)
-    connexion_app.add_api("api.spec.yaml", validate_responses=True, strict_validation=True)
+    # connexion_app.add_api("api.spec.yaml", validate_responses=True, strict_validation=True)
+    connexion_app.add_api("api.spec.yaml")
```
Comment (vkrizan) [inline drift/app.py:26]: Can we keep `strict_validation=True` and set `validate_responses=False`?

File: drift/app.py [hunk 3]
```diff
@@ -22,8 +21,9 @@ def create_app():
     """
     app_name = os.getenv("APP_NAME", "drift")
     openapi_args = {"path_prefix": config.path_prefix, "app_name": app_name}
-    connexion_app = connexion.App(__name__, specification_dir="openapi/", arguments=openapi_args)
-    connexion_app.add_api("api.spec.yaml", validate_responses=True, strict_validation=True)
+    connexion_app = FlaskApp(__name__, specification_dir="openapi/", arguments=openapi_args)
+    # connexion_app.add_api("api.spec.yaml", validate_responses=True, strict_validation=True)
```
Comment (vkrizan) [inline drift/app.py:25]: I'd recommend to remove code in the comment and rather mention that response validation is turned off.

**Review by vkrizan — APPROVED** [review]

File: tests/test_v1_api.py (hunk 1 above)
Comment (RoamingNoMaD) [reply tests/test_v1_api.py:51]: There is no constant for that and since it's only used in this one test I think it's fine to leave as is.

**Comment by petrblaho** [comment]
Superseded by https://github.com/RedHatInsights/drift-backend/pull/657

**Pull request closed without merging**